pages/api/views/[slug].js

```diff
@@ -1,25 +1,44 @@
-import db from '@/lib/firebase';
+import db from '@/lib/planetscale';
 
 export default async function handler(req, res) {
-  if (req.method === 'POST') {
-    const ref = db.ref('views').child(req.query.slug);
-    const { snapshot } = await ref.transaction((currentViews) => {
-      if (currentViews === null) {
-        return 1;
+  try {
+    if (req.method === 'POST') {
```

Comment:
Not sure if `last_insert_id()` is safe here, because on MySQL it depends on the client connection ( every connection to MySQL have a session that stores the last inserted id from that connection afaik, also i think you might get a race condition if multiple users view multiple posts.

not sure if this works or not:

```sql
update count = (select count(id) from views where slug = ?), updated_at = now();
```

check here for more info on last_insert_id() behavior with MySQL https://dev.mysql.com/doc/refman/8.0/en/information-functions.html#function_last-insert-id

Reply:
@dgraham Any thoughts here?